fml/memory/weak_ptr.h

```diff
@@ -204,7 +204,7 @@ class TaskRunnerAffineWeakPtr {
       T* ptr,
       fml::RefPtr<fml::internal::WeakPtrFlag>&& flag,
       DebugTaskRunnerChecker checker)
-      : ptr_(ptr), flag_(std::move(flag)), checker_(checker) {}
+      : ptr_(ptr), flag_(std::move(flag)), checker_(std::move(checker)) {}
```

Comment:
Why move it here but pass by const ref above?

Reply:
Ah never mind; this is a DebugTaskRunnerChecker and the other is a DebugThreadRunner. I had assumed they were the same type. Looking at the top of this file, they look somewhat analogous, differing only in whether they assert we're on the creation thread or creation task runner. Curious why the difference in how they're passed?

Should these types be move-only? I understand what these are trying to accomplish but I'm not familiar with the code in question or its history -- for some reason I had thought we were just using a threadlocal for this.

Reply:
They should not be moved, the factory should always own a checker to spawn multiple weakPtrs

Reply:
Changed it to a const reference, there isn't a good reason for this to be `std::move`d.